sql/sql_class.cc

```diff
@@ -4505,6 +4505,7 @@ void THD::inc_status_created_tmp_disk_tables()
 void THD::inc_status_created_tmp_tables()
 {
   status_var.created_tmp_tables++;
+  query_plan_flags|= QPLAN_TMP_TABLE;
```

Comment:
After the fix, this same statement in create_tmp_table becomes redundant and should be removed